client/src/components/dataset/DataTableSidebar.jsx

```diff
@@ -13,12 +14,14 @@ export default function DataTableSidebar(props) {
       return <Filter {...props} />;
     case 'transformationLog':
       return <TransformationLog {...props} />;
+    case 'combine-columns':
+      return <CombineColumns {...props} />;
     default:
       throw new Error(`Unknown sidebar type ${props.type}`);
   }
 }
 
 DataTableSidebar.propTypes = {
-  type: PropTypes.oneOf(['edit', 'filter', 'transformationLog']).isRequired,
+  type: PropTypes.oneOf(['edit', 'filter', 'transformationLog', 'combine-columns']).isRequired,
```

Comment:
We are mixing `camelCase` and `kebab-case` here. Perhaps rename to `combineColumns` (in a few places)?